radixdlt/src/integration/java/com/radixdlt/integration/distributed/simulation/MockedLedgerModule.java

```diff
@@ -30,34 +32,38 @@
 import com.radixdlt.consensus.epoch.EpochChange;
 import com.radixdlt.consensus.liveness.NextCommandGenerator;
 import com.radixdlt.consensus.sync.SyncRequestSender;
-import com.radixdlt.crypto.Hash;
+import com.radixdlt.universe.Universe;
 import com.radixdlt.consensus.PreparedCommand;
 import io.reactivex.rxjava3.core.Observable;
 
 public class MockedLedgerModule extends AbstractModule {
 	private final BFTValidatorSet validatorSet;
 
 	public MockedLedgerModule(BFTValidatorSet validatorSet) {
-		this.validatorSet = validatorSet;
+		this.validatorSet = Objects.requireNonNull(validatorSet);
 	}
 
 	@Override
 	public void configure() {
 		bind(CommittedStateSyncRx.class).toInstance(Observable::never);
 		bind(EpochChangeRx.class).toInstance(Observable::never);
-		EpochChange initialEpoch = new EpochChange(VertexMetadata.ofGenesisAncestor(validatorSet), validatorSet);
-		bind(EpochChange.class).toInstance(initialEpoch);
 		bind(NextCommandGenerator.class).toInstance((view, aids) -> null);
 		bind(SyncRequestSender.class).toInstance(req -> { });
 	}
 
+	@Provides
+	@Singleton
+	EpochChange initialEpoch(Universe universe, VertexMetadata genesisMetadata) {
+		return new EpochChange(VertexMetadata.ofGenesisAncestor(validatorSet, universe.getTimestamp()), validatorSet);
+	}
+
 	@Provides
 	@Singleton
 	Ledger syncedLedger() {
 		return new Ledger() {
 			@Override
 			public PreparedCommand prepare(Vertex vertex) {
-				return PreparedCommand.create(0, Hash.ZERO_HASH);
+				return PreparedCommand.create(0, vertex.getQC().getTimestampedSignatures().weightedTimestamp());
```

Comment:
Given that this is a mocked Ledger would prefer to see this be 0L to emphasize/prove that consensus does not depend on the result of prepare().